pyo3-ffi/src/cpython/tupleobject.rs

```diff
@@ -21,16 +23,26 @@ pub unsafe fn PyTuple_GET_SIZE(op: *mut PyObject) -> Py_ssize_t {
 }
 
 #[inline]
-#[cfg(not(any(PyPy, GraalPy)))]
 pub unsafe fn PyTuple_GET_ITEM(op: *mut PyObject, i: Py_ssize_t) -> *mut PyObject {
-    *(*(op as *mut PyTupleObject)).ob_item.as_ptr().offset(i)
+    #[cfg(not(any(GraalPy, PyPy)))]
+    return *(*(op as *mut PyTupleObject)).ob_item.as_ptr().offset(i);
+    #[cfg(any(GraalPy, PyPy))]
+    return PyTuple_GetItem(op, i);
 }
 
 /// Macro, *only* to be used to fill in brand new tuples
 #[inline]
-#[cfg(not(any(PyPy, GraalPy)))]
 pub unsafe fn PyTuple_SET_ITEM(op: *mut PyObject, i: Py_ssize_t, v: *mut PyObject) {
-    *(*(op as *mut PyTupleObject)).ob_item.as_mut_ptr().offset(i) = v;
+    #[cfg(not(any(GraalPy, PyPy)))]
+    {
+        *(*(op as *mut PyTupleObject)).ob_item.as_mut_ptr().offset(i) = v;
+    }
+    #[cfg(any(GraalPy, PyPy))]
+    {
+        let previous = PyTuple_GetItem(op, i);
+        Py_XDECREF(previous);
```

Comment:
Ditto for the two decrefs where there shouldn't be any.

Reply:
Confused myself, changed to Py_XINCREF to counter the decref in `PyTuple_SetItem`

Reply:
The memory might be unitialized, the decref in `PyTuple_SetItem` could segfault. It's unfortunate that we didn't leave any good symbol to use for this. We only have `PyTruffleTuple_GetItems`, I'm not sure we want to expose that.

Reply:
It shouldn't be uninitialized though. If it's managed, we used `calloc` to get it to native, and the native allocator does `memset(obj, '\0', size);`

Reply:
If it's a native subclass it can use its own allocator that doesn't clear the memory. And even when not, the incref afterwards is just a terrible workaround and reliance on implementation details. Could you export a proper function for this on GraalPy side and make this graalpy 25+?